src/WebJobs.Script/ManagedDependencies/ManagedDependencyProvider.cs

```diff
@@ -0,0 +1,169 @@
+                           moduleName,
+                           majorVersion,
+                           managedDependencyConfigs);
+                    }
+                }
+                //If it's Linux and app service environment
+                else if (oSPlatform == OSPlatform.Linux
+                    && !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId)))
+                {
+                    dependencyDownloadPath = managedDependencyConfigs.ManagedDependencyHomeDirectoryLinux;
+                    isDownloadRequired = IsDependencyDownloadRequired(managedDependencyConfigs.ManagedDependencyHomeDirectoryLinux,
+                        moduleName,
+                        majorVersion,
+                        managedDependencyConfigs);
+                }
+                //If it's local environment Linux or Windows
+                else
+                {
+                    dependencyDownloadPath = Path.Combine(Path.GetDirectoryName(hostFilePath), ManagedDependenciesPropertiesName);
+                    isDownloadRequired = IsDependencyDownloadRequired(dependencyDownloadPath,
+                        moduleName,
+                        majorVersion,
+                        managedDependencyConfigs);
+                }
+
+                dependency.Path = dependencyDownloadPath;
+                dependency.Isdownloadrequired = isDownloadRequired;
+                dependencyInstallationList.Add(dependency);
+            }
+
+            return dependencyInstallationList;
+        }
+
+        private bool IsDependencyDownloadRequired(string dependencyDownloadPath, string moduleName, string majorVersion, IManagedDependencyConfigs managedDependencyConfigs)
+        {
+            if (string.IsNullOrWhiteSpace(dependencyDownloadPath)
+                || string.IsNullOrWhiteSpace(moduleName)
+                || string.IsNullOrWhiteSpace(majorVersion))
+            {
+                return false;
+            }
+
+            //If the manageddependencies library doesn't exit then create the directory and module needs download
+            if (!Directory.Exists(dependencyDownloadPath))
+            {
+                Directory.CreateDirectory(dependencyDownloadPath);
+                return true;
+            }
+
+            var installedModuleInfoFilePath = Path.Combine(dependencyDownloadPath, moduleName, managedDependencyConfigs.InstalledDependencyInfoFileName);
+            if (!File.Exists(installedModuleInfoFilePath))
+            {
+                return true;
+            }
+
+            var installedModule = DesirializeJsonFile<InstalledModule>(installedModuleInfoFilePath).GetAwaiter().GetResult();
+            if (string.IsNullOrWhiteSpace(installedModule.InstalledVersion))
+            {
+                return true;
+            }
+
+            //If installed major version and the major version in host.json doesn't match then modules needs download
```

Comment:
Nit: Downlaod modules if major version is installed but does not match the version specified in host.json